string/string.cc

```diff
@@ -1,12 +1,113 @@
 #include "string.h"
 
-String::String()
+#include <algorithm>
+#include <iterator>
+#include <memory>
+#include <string>
+
+
+const char* EMPTY = "";
+
+
+size_t strlen(const char* s)
+{
+	size_t len = 0;
+
+	while (s[len] != '\0')
+		len++;
+
+	return len;
+}
+
+
+String::String() {}
```

Comment:
Or, you can just say:

    String::String() = default;

This seems more explicit to me.